build(deps): bump coredns from 1.3.1 to 1.6.2

Ebaneck opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->
'deps', 'build'

**Context**:

See: #2572  

**Summary**:

- Since the target branch runs K8s 1.16 and following the kubeadm standards as specified in the issue, let us bump the version of coredns to `1.6.2`.

- Following `kubeadm` standards for K8s 1.16 release as specified here: https://github.com/kubernetes/kubernetes/blob/release-1.16/cmd/kubeadm/app/phases/addons/dns/manifests.go#L306.

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2572

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

Ebaneck committed:
deps: update coredns to version 1.6.2

Our current deployed version for coredns is `1.3.1`
which seems to be very old.

Since we plan a minor release, we should bump the version of Coredns
to a recent release.

Note: since this branch runs K8S 1.16, we bind the coredns version
to `1.6.2` following same standards from kubeadm.

Closes: #2572

bert-e commented:
# Hello ebaneck,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2572-update-coredns` into
`development/2.5`
* [w/2.6/improvement/2572-update-coredns](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2572-update-coredns) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

NicolasT commented:
Did you check for any differences in the `Corefile` as well?

Ebaneck commented:
> Did you check for any differences in the `Corefile` as well?

Nope, I was not aware. I am going to check now

Ebaneck committed:
coredns: update the Corefile in coredns `deployed.sls`

Since we are updating coredns from 1.3.1 to 1.6.2, we need
to update the Corefile to remove deprecated plugins.

Given that we follow kubeadm standards, the Corefile is
updated using the following specifification;

```
https://github.com/kubernetes/kubernetes/blob/release-1.16/cmd/kubeadm/app/phases/addons/dns/manifests.go#L306
```

bert-e commented:
# History mismatch



Merge commit #36f55cb5fd43d5236786a6b849b9d0d1e303dae7 on the integration branch
`w/2.6/improvement/2572-update-coredns` is merging a branch which is neither the current
branch `improvement/2572-update-coredns` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2572-update-coredns` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

NicolasT commented on salt/metalk8s/kubernetes/coredns/deployed.sls:
```diff
@@ -17,15 +17,15 @@ Create coredns ConfigMap:
                 health
                 kubernetes {{ coredns.cluster_domain }} {{ coredns.reverse_cidrs }} {
                   pods insecure
-                  upstream
                   fallthrough in-addr.arpa ip6.arpa
                 }
                 prometheus :9153
-                proxy . /etc/resolv.conf
+                forward . /etc/resolv.conf
                 cache 30
                 loop
                 reload
                 loadbalance
+                ready
             }
```
It appears these changes are made using some tool? However, they're not (necessarily) in line with what `kubeadm` now deploys: https://github.com/kubernetes/kubernetes/blob/87473a5d27e52f8eb86575898cd6e6d8b8e881be/cmd/kubeadm/app/phases/addons/dns/manifests.go#L303 (note: this is `master`, should figure out what `kubeadm` does for the K8s release we want to ship).

Would make sense to re-run/update https://github.com/NicolasT/kubeadm-dry-run

Ebaneck replied:
Ideally here,  it's a 2.5 branch and we ship K8S 1.16.8.

Cross-checking with the `kubeadm` specs from a corresponding release here: https://github.com/kubernetes/kubernetes/blob/release-1.16/cmd/kubeadm/app/phases/addons/dns/manifests.go#L306, It seems we are all good. 

Note: One thing which the `kubeadm` Corefile adds is a TTL set to 30 seconds. I will add this and be done.

NicolasT replied:
I would use the same content as what `kubeadm` deploys. Indeed, `ready` is in there, but at a different location in the file. Would be easier to 'review' if we just deploy the exact same thing.

Furthermore, the RBAC resources may need to be validated as well?

Ebaneck replied:
Yeah, I followed same placement as that of `kubeadm`

Ebaneck replied:
+ all good with RBAC definitions

gdemonet replied:
> Would make sense to re-run/update https://github.com/NicolasT/kubeadm-dry-run

Definitely, even make it part of our procedure for updating K8s dependencies. Could we somehow use this as a tool for rendering our formulas, just as we do for Helm?
Also, in the future I think it'd make sense to use the [kubeadm Operator](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/kubeadm/20190916-kubeadm-operator.md) (once it is implemented) directly in the product (would make it much easier to maintain).

NicolasT replied:
Not convinced about the latter. Adding another tool which, next to Salt, does things on hosts will result in a messy situation where we no longer know what does what.

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2572-update-coredns` into
`development/2.5`
* [w/2.6/improvement/2572-update-coredns](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2572-update-coredns) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

Ebaneck commented:
Hold until after release!

gdemonet commented:
/wait

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/2572-update-coredns` has diverged from
`development/2.5` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.5` into `improvement/2572-update-coredns`
* Rebase `improvement/2572-update-coredns` onto `origin/development/2.5`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

Ebaneck committed:
Merge remote-tracking branch 'origin/development/2.5' into improvement/2572-update-coredns

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

Ebaneck commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/c54bfb54bf3fa3ddf1f28e6d30665362b3a7a73b) did not succeed in branch improvement/2572-update-coredns.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye ebaneck.
